plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -127,15 +129,40 @@ public PluginManager(Config cfg) {
         skipFailedPlugins = cfg.isSkipFailedPlugins();
         hashFunction = cfg.getHashFunction();
         httpClient = null;
+        try{
+            userAgentInformation = this.getUserAgentInformation();
+        }
+        catch(UserAgentInformationException e){
+            logVerbose("Unable to get the UserAgentInformation");
+        }
     }
 
+    private String getUserAgentInformation() throws UserAgentInformationException{
```

Comment:
this is not thrown please remove

Reply:
I see my mistake here, rather than removing it. Shall I add a `throw new UserAgentInformationException` to avoid plain IOException and then logging it? Oleg told me to do some checked exceptions, I feel this way both criteria's will be fulfilled what do you say?

Reply:
No need, if you can't load it just fallback to a default user agent

Reply:
Okay, Alright then.